sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -126,24 +131,31 @@ public static ClientPipeline Create(ClientPipelineOptions options, params Pipeli
         }
         else
         {
-            // TODO: Transport implementation will come in a later PR.
-            //// Add default transport.
-            //policies[index++] = HttpClientPipelineTransport.Shared;
+            // Add default transport.
+            policies[index++] = HttpClientPipelineTransport.Shared;
         }
 
-        return new ClientPipeline(policies, perCallIndex, perTryIndex, beforeTransportIndex); ;
+        return new ClientPipeline(policies,
+            options.NetworkTimeout ?? DefaultNetworkTimeout,
+            perCallIndex, perTryIndex, beforeTransportIndex);
     }
 
+    #endregion
+
     public PipelineMessage CreateMessage() => _transport.CreateMessage();
 
     public void Send(PipelineMessage message)
     {
+        message.NetworkTimeout ??= _networkTimeout;
```

Comment:
I'm wondering if setting this on the message could cause confusion for protocol method scenarios in which we are mutating the message provided by the developer in a non-obvious way.

Reply:
Can you say more about the specific case you're concerned about?  My thinking is that it's an internal API that's nullable, so it can't be set directly by either an end-user or a client-author.  We'll set it internal from RequestOptions when we add that type -- if an end-user specifies it -- and if it's nullable, it'll only be overwritten by the client-scoped value if it hasn't been set in RequestOptions, which would take precedence over the pipeline value (per-invocation taking precedence over per-client).  But if there's a case I'm missing, please help me understand and will address!

Reply:
I was assuming this was [HttpMesssage.NetworkTimeout](https://github.com/Azure/azure-sdk-for-net/blob/c61d44db4c101ab7d24095465519bfd688c10b7d/sdk/core/Azure.Core/src/HttpMessage.cs#L84) which is public and settable by end-users

Reply:
Related to the discussion here: https://github.com/Azure/azure-sdk-for-net/pull/41223#discussion_r1450993518

Offline @KrzysztofCwalina has suggested that we make BufferReponse public and remove the static setters from the policy/transport.  The concern about this (and reason we didn't do it initially) is that if the response buffering policy is not present in the pipeline, end users can set BufferResponse to true and it will have no effect on the message.  This feels like a pretty big design problem in our pipeline.

Depending on where we land with BufferResponse, I think NetworkTimeout should do the same thing.  For me, it's even more straightforward than BufferResponse because a good default pipeline should implement a network timeout regardless -- this feels like a pretty strong argument for moving that functionality out of the response buffering policy one way or another.  If we do make it public, I still don't like having two ways to set a network timeout per-invocation (the other is on RequestOptions), but I guess you can make the argument that one is for end-users and one is for client-authors, so maybe at the end of the day it's ok.

Reply:
Ok, we will make these properties public on message per @KrzysztofCwalina:
_"I would just make the property public, not protected, regardless of what happens with the buffering policy. The argument being: we would not be fixing anything in Azure.Core regardless as we already have the APIs that let people make the buffering not work correctly. Now, this issue would be worth fixing in CM, if the fix does not result in a franken design, but because of the message inheritance, it does, and so I would clasify it as: we have a problem, there is no cleam fix (besides moving buffering to the transport), and because there is no clean fix, we should not be trading one problem for another"_

Reply:
In that case, @christothes, I think setting it in pipeline.Send is still ok because we're only setting it if it hasn't been set by an end-user.  We're effectively saying -- if there's no per-invocation override coming from either an end-user or a client-author, the ClientModel internals will ensure that pipeline policies will get the PipelineOptions value (passed by the end-user) as the network timeout value they observe.

@christothes, let me know if that doesn't resolve your concern, and we can continue to discuss!

Reply:
My general thought here is that we should avoid mutating state owned by the end-user. Even though in this case the end result is the same as expected, the implementation details leaks into the user controlled message. This could affect how user controlled policies interact with the message, for instance.

Perhaps this is an acceptable trade-off considering the alternatives, but I'm not super familiar with the design trade-offs that have already been discussed here.

Reply:
I think that making the property publicly settable implies that CM should not muck with it. Could CM avoid setting the public property and instead read the default value from the pipeline options in ResponseBufferingPolicy?

Reply:
> This seems like it would only be true for convenience methods, not protocol methods.

Protocol methods don't expose message ... or am I missing your meaning, @christothes?

Reply:
> I was thinking of examples we have where we demonstrate using the pipeline directly. But perhaps I am confusing concepts here.

Ah, yes, I don't think of that as using a protocol method (since no protocol method is involved), but you are right we expose this on DPG clients.

In that case, though, I don't think we have a problem either.  The end-user either sets or doesn't set a NetworkTimeout value on the message.  There is no "client-author" involved, and so in pipeline.Send, if the value is set, the end-user value is respected.  If the value is not set and the network timeout value was passed in options when the pipeline was constructed, the pipeline will respect the pipeline options value.  If the value is not set and no value was passed in options, it will use the default network timeout value.

Can you help me understand where the problem is in this logic?

Reply:
> I don't have an issue with the precedence rules, but rather that we are changing what feels like it should be user supplied data. The precedence can remain the same without changing the property, I think.

I'm not sure this is true.  A custom policy can't observe the network timeout that was passed to the pipeline via options without it being set on message.

Reply:
> Where are we advertising this contract? The property itself is nullable.

Where would you like to advertise it?  We can add comments in the code, or in the refdocs.  Using nullable properties to enforce precedence rules is a relatively common pattern.  That way you can know whether or not a user provided a value.

Reply:
The custom policy should have access to the custom network timeout, no? I don't think network timeout is meant for client authors to change.

Reply:
> Where would you like to advertise it? We can add comments in the code, or in the refdocs. Using nullable properties to enforce precedence rules is a relatively common pattern. That way you can know whether or not a user provided a value.

No issues with the property being nullable. My issue is with us saying there is a contract that even though the property is nullable, it won't ever be null because we will always set it to some default value if the user doesn't set it.

Reply:
> The custom policy should have access to the custom network timeout, no? I don't think network timeout is meant for client authors to change.

I think we may have different understandings of what's happening.  I want to make sure your concern is addressed, so let's see if we can get clarity on the concerns here and make sure we're both understanding each others' cases.

I _believe_ that if a custom policy is added to the pipeline, and the end-user sets message.NetworkTimeout, the call in pipeline.Send will not overwrite the message.NetworkTimeout value, and so it will have the end-users value when the message gets to the custom policy.

Is that the case you are describing?  If not, would you be willing to provide a code sample of the case you are concerned about and how it fails?

Reply:
> No issues with the property being nullable or with the precedence rules. My issue is with us saying there is a contract that even though the property is nullable, it won't ever be null because we will always set it to some default value if the user doesn't set it.

But it is nullable exactly because we want to know whether or not the user set it, so we know whether or not it is safe to set the value to the pipeline options value.

Reply:
> I _believe_ that if a custom policy is added to the pipeline, and the end-user sets message.NetworkTimeout, the call in pipeline.Send will not overwrite the message.NetworkTimeout value, and so it will have the end-users value when the message gets to the custom policy.

I understand and agree with that behavior. An example that I'm thinking of is a case where the user policy does NOT set the value, but instead reads the value after the request is sent. The policy would have no way of knowing where this value was set given it could have been set by both CM and a different custom policy.

> But it is nullable exactly because we want to know whether or not the user set it, so we know whether or not it is safe to set the value to the pipeline options value.

I agree that it should be nullable. I don't agree that we need to set it if the user leaves it null. The pattern I've seen with nullable options is that we document what the default (null) behavior is (in this case, the PipelineOptions setting), but that we don't actually mutate the data to set it.

Reply:
It feels like we have more to discuss here.  @JoshLove-msft, @christothes, I have filed a P0 (release blocking) issue to track this work.  Please let me know if you feel we should _also_ block the PR on resolution of this issue, or if you are ok revisiting it after we have all of the ClientModel code moved in to main from the feature branch.  Thanks!

Reply:
I'm fine with unblocking this PR